src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt

```diff
@@ -49,11 +50,13 @@ class DetailController(private val detailService: DetailService) {
   fun getDetailsByUserWithRegion(
     @PathVariable @Schema(
       description = "the number of the required region",
-      allowableValues = arrayOf("1", "2", "3", "4", "5", "6")
+      allowableValues = ["1", "2", "3", "4", "5", "6"]
     ) region: Int,
     @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) from: LocalDate,
     @RequestParam @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) to: LocalDate
   ): ResponseEntity<Collection<DetailDto>> {
+    RegionContext.setRegion(region.toString())
```

Comment:
region needs setting before the transaction begins

Reply:
not really understanding what is going on here - seems very weird to have to remember to set the region in this way before database calls.  It is almost like it should happen automatically / throw exception if it isn't then set.
The next service call then passes in the region too - is that not then required?

Reply:
ah i missed this one 👍 , will remove